toolchain/sem_ir/typed_insts.h

```diff
@@ -587,15 +587,15 @@ struct VarStorage {
 
 // HasParseNode<T> is true if T has a `Parse::Node parse_node` field.
 template <typename T, typename ParseNodeType = Parse::Node T::*>
-constexpr bool HasParseNode = false;
+static constexpr bool HasParseNode = false;
 template <typename T>
-constexpr bool HasParseNode<T, decltype(&T::parse_node)> = true;
+static constexpr bool HasParseNode<T, decltype(&T::parse_node)> = true;
 
 // HasTypeId<T> is true if T has a `TypeId type_id` field.
 template <typename T, typename TypeIdType = TypeId T::*>
-constexpr bool HasTypeId = false;
+static constexpr bool HasTypeId = false;
 template <typename T>
-constexpr bool HasTypeId<T, decltype(&T::type_id)> = true;
+static constexpr bool HasTypeId<T, decltype(&T::type_id)> = true;
```

Comment:
The linter is wrong here, and `static` is an ODR trap. Does adding `inline` instead silence the lint? If not, let's just suppress it here.

```suggestion
// HasParseNode<T> is true if T has a `Parse::Node parse_node` field.
template <typename T, typename ParseNodeType = Parse::Node T::*>
inline constexpr bool HasParseNode = false;
template <typename T>
inline constexpr bool HasParseNode<T, decltype(&T::parse_node)> = true;

// HasTypeId<T> is true if T has a `TypeId type_id` field.
template <typename T, typename TypeIdType = TypeId T::*>
inline constexpr bool HasTypeId = false;
template <typename T>
inline constexpr bool HasTypeId<T, decltype(&T::type_id)> = true;
```

Reply:
Done, works.